ecosystem/README_zh-CN.md

```diff
@@ -244,3 +244,54 @@ LlamaIndex 是一个用于构建上下文增强型 LLM 应用程序的框架。
 它选择 ollama 作为 LLM 推理引擎。你可以在[入门教程（本地模型）](<(https://docs.llamaindex.ai/en/stable/getting_started/starter_example_local/)>)中找到示例。
 
 因此，如果能够按照 [ollama 章节](#ollama)使用 ollama 部署浦语模型，你就可以顺利地将浦语模型集成到 LlamaIndex 中。
+
+### [LazyLLM](https://github.com/LazyAGI/LazyLLM)
+
+LazyLLM 是一个的低代码构建多 Agent 大模型应用的开发工具，相比于 LangChain 和 LLamaIndex，其具有极高的灵活性和易用性。
+
+当您依次通过 `pip3 install lazyllm` 和 `lazyllm install standard` 安装了 LazyLLM 之后, 您可以使用如下代码以极低的成本，基于 InternLM 搭建 chatbots，无论推理还是微调，您都无需考虑对话模型的特殊 token（如`<|im_start|>system`和`<|im_end|>`等 ）。不用担心没有权重文件，只要您能联网，下面的代码将会自动帮您下载权重文件并部署服务，您只需尽情享受 LazyLLM 给您带来的便利。
+
+```python
+from lazyllm import TrainableModule, WebModule
+m = TrainableModule('internlm2_5-7b-chat')
+# will launch a chatbot server
+WebModule(m).start().wait()
+```
+
+如果您需要进一步微调模型，可以参考如下代码。当 `TrainableModule` 的 `trainset` (数据集需下载到本地，例如：[
+alpaca_gpt4_zh](https://huggingface.co/datasets/llamafactory/alpaca_gpt4_zh))被设置之后，在调用 `WebModule` 的 `update` 函数时，会自动微调 `TrainableModule`，然后对 `TrainableModule` 和 `WebModule` 分别进行部署。
+
+```python
+from lazyllm import TrainableModule, WebModule
+m = TrainableModule('internlm2-chat-7b').trainset('/patt/to/your_data.json').mode('finetune')
+WebModule(m).update().wait()
+```
+值的一提的是，无论您用 InternLM 系列的任何一个模型，都可以使用 LazyLLM 进行推理和微调，您都无需考虑模型的切分策略，也无需考虑模型的特殊 token。<br>
+如果您想搭建自己的 RAG 应用，那么您无需像使用 LangChain 一样先启动服务推理服务，再配置 ip 和端口去启动应用程序。参考如下代码，您可以借助 LazyLLM，使用 InternLM 系列的模型，十行代码搭建高度定制的 RAG 应用，且附带文档管理服务（文档需指定本地绝对路径，可从这里下载：[rag_master](https://huggingface.co/datasets/Jing0o0Xin/rag_master)）：
+
+<details>
+<summary>点击获取import和prompt</summary>
+
+```python
+import os
+import lazyllm
+from lazyllm import pipeline, parallel, bind, SentenceSplitter, Document, Retriever, Reranker
+
+prompt = '你将扮演一个人工智能问答助手的角色，完成一项对话任务。在这个任务中，你需要根据给定的上下文以及问题，给出你的回答。'
+```
+</details>
+
+```python
+documents = Document(dataset_path='/file/to/yourpath', embed=lazyllm.TrainableModule('bge-large-zh-v1.5'))
```

Comment:
So weird. I download rag_master dataset first under /home/zhulin1/rag_master
```
(testall) [zhulin1@SH-IDC1-10-140-0-187 rag_master]$ ls
README.md  中庸.txt  大学.txt  心经.txt  论语.txt  道德经.txt  阴符经.txt
(testall) [zhulin1@SH-IDC1-10-140-0-187 rag_master]$ pwd
/home/zhulin1/rag_master
```

then run this python script. 
the file under rag_master is changed into
```
(testall) [zhulin1@SH-IDC1-10-140-0-187 sources]$ pwd
/home/zhulin1/rag_master/default/__data/sources
(testall) [zhulin1@SH-IDC1-10-140-0-187 sources]$ ls
README.md  中庸.txt  大学.txt  心经.txt  论语.txt  道德经.txt  阴符经.txt
```
and have error like this:

```
INFO] model link found at /home/zhulin1/.lazyllm/model/bge-large-zh-v1.5
[INFO] model link found at /home/zhulin1/.lazyllm/model/internlm2_5-1_8b-chat
44145: 2024-09-06 13:09:18 lazyllm INFO: (lazyllm.launcher:107) Command: /home/zhulin1/miniconda3/envs/testall/bin/python /home/zhulin1/miniconda3/envs/testall/lib/python3.10/site-packages/lazyllm/components/deploy/relay/server.py --open_port=34633  
44145: 2024-09-06 13:09:21 lazyllm INFO: (lazyllm.launcher:152) 44409: INFO:     Started server process [44409]
44145: 2024-09-06 13:09:21 lazyllm INFO: (lazyllm.launcher:152) 44409: INFO:     Waiting for application startup.
44145: 2024-09-06 13:09:21 lazyllm INFO: (lazyllm.launcher:152) 44409: INFO:     Application startup complete.
44145: 2024-09-06 13:09:21 lazyllm INFO: (lazyllm.launcher:152) 44409: INFO:     Uvicorn running on http://0.0.0.0:34633 (Press CTRL+C to quit)
44145: 2024-09-06 13:09:21 lazyllm INFO: (lazyllm.components.deploy.base:59) Capture startup message: INFO:     Uvicorn running on http://0.0.0.0:34633 (Press CTRL+C to quit)

44145: 2024-09-06 13:09:21 lazyllm INFO: (lazyllm.launcher:107) Command: /home/zhulin1/miniconda3/envs/testall/bin/python /home/zhulin1/miniconda3/envs/testall/lib/python3.10/site-packages/lazyllm/components/deploy/relay/server.py --open_port=34858  
44145: 2024-09-06 13:09:24 lazyllm INFO: (lazyllm.launcher:152) 44613: INFO:     Started server process [44613]
44145: 2024-09-06 13:09:24 lazyllm INFO: (lazyllm.launcher:152) 44613: INFO:     Waiting for application startup.
44145: 2024-09-06 13:09:24 lazyllm INFO: (lazyllm.launcher:152) 44613: INFO:     Application startup complete.
44145: 2024-09-06 13:09:24 lazyllm INFO: (lazyllm.launcher:152) 44613: INFO:     Uvicorn running on http://0.0.0.0:34858 (Press CTRL+C to quit)
44145: 2024-09-06 13:09:24 lazyllm INFO: (lazyllm.components.deploy.base:59) Capture startup message: INFO:     Uvicorn running on http://0.0.0.0:34858 (Press CTRL+C to quit)

44145: 2024-09-06 13:09:24 lazyllm ERROR: (lazyllm.flow.flow:249) An error occored when invoking `<class 'lazyllm.flow.flow._FuncWrap'>(<Function type=_work>)` with input `()` and kw `{}`
44145: 2024-09-06 13:09:24 lazyllm ERROR: (lazyllm.flow.flow:249) An error occored when invoking `<class 'lazyllm.flow.flow.Pipeline'>(<Flow type=Pipeline items=[]>
 └- <Function type=_work>
)` with input `()` and kw `{}`
Traceback (most recent call last):
  File "/home/zhulin1/miniconda3/envs/testall/lib/python3.10/site-packages/requests/models.py", line 974, in json
    return complexjson.loads(self.text, **kwargs)
  File "/home/zhulin1/miniconda3/envs/testall/lib/python3.10/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/home/zhulin1/miniconda3/envs/testall/lib/python3.10/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/home/zhulin1/miniconda3/envs/testall/lib/python3.10/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/zhulin1/test_lazyllm_rag.py", line 16, in <module>
    lazyllm.WebModule(ppl, port=23456).start().wait()
  File "/home/zhulin1/miniconda3/envs/testall/lib/python3.10/site-packages/lazyllm/module/module.py", line 305, in start
    return self._update(mode=['server'], recursive=True)
  File "/home/zhulin1/miniconda3/envs/testall/lib/python3.10/site-packages/lazyllm/tools/webpages/webmodule.py", line 390, in _update
    super(__class__, self)._update(mode=mode, recursive=recursive)
  File "/home/zhulin1/miniconda3/envs/testall/lib/python3.10/site-packages/lazyllm/module/module.py", line 250, in _update
    Parallel.sequential(*deploy_tasks)()
  File "/home/zhulin1/miniconda3/envs/testall/lib/python3.10/site-packages/lazyllm/flow/flow.py", line 204, in __call__
    output = self._run(args[0] if len(args) == 1 else package(args), **kw)
  File "/home/zhulin1/miniconda3/envs/testall/lib/python3.10/site-packages/lazyllm/flow/flow.py", line 485, in _run
    r = package(self.invoke(it, inp, **kw) for it, inp in zip(items, inputs))
  File "/home/zhulin1/miniconda3/envs/testall/lib/python3.10/site-packages/lazyllm/common/common.py", line 85, in __new__
    return super(__class__, cls).__new__(cls, args[0])
  File "/home/zhulin1/miniconda3/envs/testall/lib/python3.10/site-packages/lazyllm/flow/flow.py", line 485, in <genexpr>
    r = package(self.invoke(it, inp, **kw) for it, inp in zip(items, inputs))
  File "/home/zhulin1/miniconda3/envs/testall/lib/python3.10/site-packages/lazyllm/flow/flow.py", line 247, in invoke
    return it(__input, **kw)
  File "/home/zhulin1/miniconda3/envs/testall/lib/python3.10/site-packages/lazyllm/flow/flow.py", line 204, in __call__
    output = self._run(args[0] if len(args) == 1 else package(args), **kw)
  File "/home/zhulin1/miniconda3/envs/testall/lib/python3.10/site-packages/lazyllm/flow/flow.py", line 324, in _run
    output = self.invoke(it, output, bind_args_source=bind_args_source, **kw)
  File "/home/zhulin1/miniconda3/envs/testall/lib/python3.10/site-packages/lazyllm/flow/flow.py", line 245, in invoke
    return it(*__input, **kw) if isinstance(__input, package) else it(**__input, **kw)
  File "/home/zhulin1/miniconda3/envs/testall/lib/python3.10/site-packages/lazyllm/flow/flow.py", line 21, in __call__
    def __call__(self, *args, **kw): return self.f(*args, **kw)
  File "/home/zhulin1/miniconda3/envs/testall/lib/python3.10/site-packages/lazyllm/tools/rag/web.py", line 272, in _work
    self.demo = self.web_ui.create_ui()
  File "/home/zhulin1/miniconda3/envs/testall/lib/python3.10/site-packages/lazyllm/tools/rag/web.py", line 94, in create_ui
    del_select_group = gr.Dropdown(self.list_groups(), label="选择分组")
  File "/home/zhulin1/miniconda3/envs/testall/lib/python3.10/site-packages/lazyllm/tools/rag/web.py", line 56, in list_groups
    return response.json()["data"]
  File "/home/zhulin1/miniconda3/envs/testall/lib/python3.10/site-packages/requests/models.py", line 978, in json
    raise RequestsJSONDecodeError(e.msg, e.doc, e.pos)
requests.exceptions.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```


@JingofXin do you have any suggestion about this error?